twake/backend/node/src/core/platform/services/push/connectors/fcm/service.ts

```diff
@@ -0,0 +1,47 @@
+import { option } from "yargs";
+import { logger } from "../../../../framework";
+import { PushConfiguration, PushMessageNotification, PushMessageOptions } from "../../types";
+import { PushConnector } from "../connector";
+
+export default class FcmPushConnector implements PushConnector {
+  name = "FcmPushConnector";
+
+  constructor(readonly configuration: PushConfiguration["fcm"]) {}
+
+  async push(
+    devices: string[],
+    notification: PushMessageNotification,
+    options: PushMessageOptions,
+  ): Promise<void> {
+    const firebaseEndpoint = this.configuration.endpoint;
+    const firebaseApiKey = this.configuration.key;
+
+    for (const deviceIdentifier of devices) {
+      const pushMessage = {
+        data: {
+          notification_data: options.notification_data,
+          click_action: "FLUTTER_NOTIFICATION_CLICK",
+        },
+        notification: { ...notification, click_action: "FLUTTER_NOTIFICATION_CLICK" },
+        collapse_key: options.collapse_key,
+        registration_ids: [deviceIdentifier],
+      };
+
+      //Push to fcm
+      try {
+        const response = await (
+          await fetch(firebaseEndpoint, {
+            headers: {
+              "Content-Type": "application/json",
+              Authorization: `key=${firebaseApiKey}`,
+            },
+            body: JSON.stringify(pushMessage),
+          })
+        ).json();
+        logger.info(`${this.name} - Reply from FCM: ${JSON.stringify(response)}`);
```

Comment:
Should be `debug` when printing JSON. Also, `json()` on fetch response can take time (printing is also something which costs)

Reply:
Ok, logged only if response was not 200 now